src/payment/strategies/paypal-commerce/paypal-commerce-payment-processor.ts

```diff
@@ -0,0 +1,62 @@
+import { MissingDataError, MissingDataErrorType } from '../../../common/error/errors';
+
+import { OverlayPaypal } from './index';
+
+const modalWidth = 450;
+const modalHeight = 600;
+
+export default class PaypalCommercePaymentProcessor {
+    private _window = window;
+    private _popup?: Window | null;
+
+    constructor(
+        private _overlay: OverlayPaypal
+    ) {}
+
+    paymentPayPal(approveUrl: string): Promise<any> {
+        return new Promise((resolve, reject) => {
+            const closeWindow = (isResolve: boolean) => {
+                this._overlay.remove();
+
+                if (this._popup) {
+                    this._popup.close();
+                }
+
+                isResolve
+                    ? resolve(true)
+                    : reject(new MissingDataError(MissingDataErrorType.MissingPayment));
+            };
+
+            const handlerMessage = (event: MessageEvent) => {
+                if (event.origin.includes('sandbox.paypal.com')) {
+                    const data = JSON.parse(event.data);
+
+                    if (data.operation === 'return_to_merchant' && data.updateParent) {
+                        this._window.removeEventListener('message', handlerMessage);
+                        closeWindow(true);
+                    }
+                }
+            };
+
+            this._window.addEventListener('message', handlerMessage);
+            const paramsModalWindow = `
```

Comment:
maybe you could define this in a more readable way?
modalWindowParams = [
  [ 'left',  Math.round((window.screen.height - modalWidth) / 2) ],
  [ 'top', Math.round((window.screen.width - modalHeight) / 2) ],
  // ...
].map(value => value.join('=')).join(',');

Reply:
Hm, I'm not sure. For me it's quite readable. And in your case array was created, after this was iterated over and joined. 
It's so costly for just one string